core/src/main/java/org/elasticsearch/index/mapper/StringFieldMapper.java

```diff
@@ -168,7 +168,9 @@ public StringFieldMapper build(BuilderContext context) {
                 }
             }
             if (positionIncrementGap != POSITION_INCREMENT_GAP_USE_ANALYZER) {
-                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0) {
+                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0 &&
+                        context.indexCreatedVersion() != null &&
```

Comment:
created version should not be able to be null. We have numerous checks that depend on it in the base FieldMapper.